internal/config/config.go

```diff
@@ -661,6 +659,27 @@ func bgpAdvertisementsFromLegacyCR(ads []metallbv1beta1.LegacyBgpAdvertisement,
 	return ret, nil
 }
 
+func (ad *BGPAdvertisement) communtiesForBGPAd(adCommunities []string, communities map[string]uint32) error {
+	for _, c := range adCommunities {
+		v, err := ParseCommunity(c)
+		if err != nil {
+			if strings.Contains(err.Error(), "invalid community string") {
+				// Check if the community exists.
+				if v, ok := communities[c]; ok {
+					ad.Communities[v] = true
+					continue
+				} else {
+					return TransientError{fmt.Sprintf("No such community exists: %s", err)}
+				}
+			} else {
```

Comment:
I'd move this on top and return early, will save another else.

Reply:
done